modules/memberships/class-jetpack-memberships.php

```diff
@@ -0,0 +1,283 @@
+	 * @see hook 'rest_api_allowed_post_types'
+	 * @return array
+	 */
+	public function allow_rest_api_types( $post_types ) {
+		$post_types[] = self::$post_type_plan;
+
+		return $post_types;
+	}
+
+	/**
+	 * Allows custom meta fields to sync.
+	 *
+	 * @param array $post_meta - previously changet post meta.
+	 *
+	 * @return array
+	 */
+	public function allow_sync_post_meta( $post_meta ) {
+		$meta_keys = array_map(
+			array( $this, 'return_meta' ),
+			$this->get_plan_property_mapping()
+		);
+		return array_merge( $post_meta, array_values( $meta_keys ) );
+	}
+
+	public function return_meta( $map ) {
+		return $map['meta'];
+	}
+	/**
+	 * Callback that parses the membership purchase shortcode.
+	 *
+	 * @param array       $attrs - attributes in the shortcode. `id` here is the CPT id of the plan.
+	 * @param string|bool $content - needed for the callback.
+	 *
+	 * @return string|void
+	 */
+	public function render_button( $attrs ) {
+		Jetpack_Gutenberg::load_assets_as_required( self::$button_block_name);
+
+		if ( empty( $attrs['planId'] ) ) {
+			return;
+		}
+		$id = $attrs['planId'];
+		$product = get_post( $id );
+		if ( ! $product || is_wp_error( $product ) ) {
+			return;
+		}
+		if ( $product->post_type !== self::$post_type_plan || 'trash' === $product->post_status ) {
+			return;
+		}
+		$plan = self::product_post_to_array( $product );
+		$data = array(
+			'blog_id' => $this->get_blog_id(),
+			'id'	  => $id,
+			'button_label' => sprintf( __( '$%s Contribution', 'jetpack' ),$this->format_price( $plan ) ),
+			'powered_text' => __( 'Powered by WordPress.com', 'jetpack' ),
+		);
+
+		$classes = array(
+			'components-button',
+			'is-primary',
+			'is-button',
+			'wp-block-jetpack-' . self::$button_block_name,
+			self::$css_classname_prefix . '-' . $data['id'],
+		);
+		if ( isset( $attrs['className'] ) ) {
+			array_push( $classes, $attrs['className'] );
+		}
+		if ( isset( $attrs['submitButtonText'] ) ) {
+			$data['button_label'] = $attrs['submitButtonText'];
+		}
+		$button_styles = array();
+		if ( ! empty( $attrs['customBackgroundButtonColor'] ) ) {
+			array_push(
+				$button_styles,
+				sprintf(
+					'background-color: %s',
+					sanitize_hex_color( $attrs['customBackgroundButtonColor'] )
+				)
+			);
+		}
+		if ( ! empty( $attrs['customTextButtonColor'] ) ) {
+			array_push(
+				$button_styles,
+				sprintf(
+					'color: %s',
+					sanitize_hex_color( $attrs['customTextButtonColor'] )
+				)
+			);
+		}
+		$button_styles = implode( $button_styles, ';' );
+		add_thickbox();
+		return sprintf(
+			'<button data-blog-id="%d" data-powered-text="%s" data-plan-id="%d" class="%s" style="%s">%s</button>',
+			$data['blog_id'],
+			$data['powered_text'],
+			$data['id'],
+			implode( $classes, ' ' ),
+			$button_styles,
+			$data['button_label']
+		);
+	}
+
+	private function get_blog_id() {
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			return get_current_blog_id();
+		}
+
+		return Jetpack_Options::get_option( 'id' );
```

Comment:
Not a change for this PR, but just noting that feels weird that folks need to do `IS_WPCOM` for a simple thing like this — could the wpcom check be in `get_option` method directly? @jeherve